rpc/estimate_fee.go

```diff
@@ -37,10 +37,45 @@ type FeeEstimate struct {
 	Unit            *FeeUnit   `json:"unit,omitempty"`
 }
 
+type FeeEstimate struct {
+	L1GasConsumed     *felt.Felt `json:"l1_gas_consumed,omitempty"`
+	L1GasPrice        *felt.Felt `json:"l1_gas_price,omitempty"`
+	L2GasConsumed     *felt.Felt `json:"l2_gas_consumed,omitempty"`
+	L2GasPrice        *felt.Felt `json:"l2_gas_price,omitempty"`
+	L1DataGasConsumed *felt.Felt `json:"l1_data_gas_consumed,omitempty"`
+	L1DataGasPrice    *felt.Felt `json:"l1_data_gas_price,omitempty"`
+	OverallFee        *felt.Felt `json:"overall_fee"`
+	Unit              *FeeUnit   `json:"unit,omitempty"`
+}
+
 /****************************************************
 		Estimate Fee Handlers
 *****************************************************/
 
+func FeeEstimateToV0_7(feeEstimate FeeEstimate) FeeEstimateV0_7 {
+	return FeeEstimateV0_7{
+		GasConsumed:     feeEstimate.L1GasConsumed,
+		GasPrice:        feeEstimate.L1GasPrice,
+		DataGasConsumed: feeEstimate.L1DataGasConsumed,
+		DataGasPrice:    feeEstimate.L1DataGasPrice,
+		OverallFee:      feeEstimate.OverallFee,
+		Unit:            feeEstimate.Unit,
+	}
+}
+
+func (h *Handler) EstimateFeeV0_7(broadcastedTxns []BroadcastedTransaction,
+	simulationFlags []SimulationFlag, id BlockID,
+) ([]FeeEstimateV0_7, http.Header, *jsonrpc.Error) {
+	result, httpHeader, err := h.simulateTransactions(id, broadcastedTxns, append(simulationFlags, SkipFeeChargeFlag), true)
+	if err != nil {
+		return nil, httpHeader, err
+	}
+
+	return utils.Map(result, func(tx SimulatedTransaction) FeeEstimateV0_7 {
```

Comment:
you can pass function name (=pointer) like:
```
return utils.Map(result, FeeEstimateToV0_7)
```

Reply:
I don't think so, `feeEstimateToV0_7` adapts `FeeEstimate`, not `SimulatedTransaction`. 
`return utils.Map(result, feeEstimateToV0_7)` requires changes in `feeEstimateToV0_7`.